<a href="https://colab.research.google.com/github/kitiluck/BSC_DPDM23/blob/main/Freq_HW1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import os

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [ ]:
from google.colab import drive
drive.mount('/content/drive')

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
from google.colab import drive
drive.mount('/content/drive')
path = '/content/drive/MyDrive/bsc_dpdm2023/reduced_marketbasket.csv'

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
data = pd.read_csv(os.path.join(path,'/content/drive/MyDrive/bsc_dpdm2023/reduced_marketbasket.csv'),encoding='latin-1')
data

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


InvoiceNo StockCode                          Description  Quantity  \
0         536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1         536365     71053                  WHITE METAL LANTERN         6   
2         536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3         536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4         536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
...          ...       ...                                  ...       ...   
541904    581587     22613          PACK OF 20 SPACEBOY NAPKINS        12   
541905    581587     22899         CHILDREN'S APRON DOLLY GIRL          6   
541906    581587     23254        CHILDRENS CUTLERY DOLLY GIRL          4   
541907    581587     23255      CHILDRENS CUTLERY CIRCUS PARADE         4   
541908    581587     22138        BAKING SET 9 PIECE RETROSPOT          3   

             InvoiceDate  UnitPrice  CustomerID         Country  
0       12/01/2010 08:26       2.55     17850.0  United Kingdom  
1       12/01/2010 08:26       3.39     17850.0  United Kingdom  
2       12/01/2010 08:26       2.75     17850.0  United Kingdom  
3       12/01/2010 08:26       3.39     17850.0  United Kingdom  
4       12/01/2010 08:26       3.39     17850.0  United Kingdom  
...                  ...        ...         ...             ...  
541904  12/09/2011 12:50       0.85     12680.0          France  
541905  12/09/2011 12:50       2.10     12680.0          France  
541906  12/09/2011 12:50       4.15     12680.0          France  
541907  12/09/2011 12:50       4.15     12680.0          France  
541908  12/09/2011 12:50       4.95     12680.0          France  

[541909 rows x 8 columns]

In [ ]:
data #check data

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


InvoiceNo StockCode                          Description  Quantity  \
0         536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1         536365     71053                  WHITE METAL LANTERN         6   
2         536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3         536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4         536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
...          ...       ...                                  ...       ...   
541904    581587     22613          PACK OF 20 SPACEBOY NAPKINS        12   
541905    581587     22899         CHILDREN'S APRON DOLLY GIRL          6   
541906    581587     23254        CHILDRENS CUTLERY DOLLY GIRL          4   
541907    581587     23255      CHILDRENS CUTLERY CIRCUS PARADE         4   
541908    581587     22138        BAKING SET 9 PIECE RETROSPOT          3   

             InvoiceDate  UnitPrice  CustomerID         Country  
0       12/01/2010 08:26       2.55     17850.0  United Kingdom  
1       12/01/2010 08:26       3.39     17850.0  United Kingdom  
2       12/01/2010 08:26       2.75     17850.0  United Kingdom  
3       12/01/2010 08:26       3.39     17850.0  United Kingdom  
4       12/01/2010 08:26       3.39     17850.0  United Kingdom  
...                  ...        ...         ...             ...  
541904  12/09/2011 12:50       0.85     12680.0          France  
541905  12/09/2011 12:50       2.10     12680.0          France  
541906  12/09/2011 12:50       4.15     12680.0          France  
541907  12/09/2011 12:50       4.15     12680.0          France  
541908  12/09/2011 12:50       4.95     12680.0          France  

[541909 rows x 8 columns]

In [ ]:
data.groupby(['Country']).sum() #ดูข้อมูลมาจากกี่ประเทศ

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)
<ipython-input-322-25f447ef123a>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  data.groupby(['Country']).sum() #ดูข้อมูลมาจากกี่ประเทศ


Quantity    UnitPrice    CustomerID
Country                                                  
Australia                83653     4054.750  1.569300e+07
Austria                   4827     1701.520  5.021102e+06
Bahrain                    260       86.570  2.100270e+05
Belgium                  23152     7540.130  2.571829e+07
Brazil                     356      142.600  4.086080e+05
Canada                    2763      910.580  2.615483e+06
Channel Islands           9479     3738.550  1.128522e+07
Cyprus                    6317     3920.070  7.715880e+06
Czech Republic             592       88.150  3.834300e+05
Denmark                   8188     1266.950  4.876734e+06
EIRE                    142637    48447.190  1.103917e+08
European Community         497      294.050  9.215880e+05
Finland                  10666     3786.850  8.699324e+06
France                  110480    43031.990  1.076489e+08
Germany                 117448    37666.000  1.200751e+08
Greece                    1556      713.290  2.008584e+06
Hong Kong                 4769    12241.500  0.000000e+00
Iceland                   2458      481.210  2.247154e+06
Israel                    4353     1079.040  3.164467e+06
Italy                     7999     3879.390  1.015666e+07
Japan                    25218      814.860  4.567292e+06
Lebanon                    386      242.440  5.743800e+05
Lithuania                  652       99.440  5.366200e+05
Malta                      944      666.010  2.158496e+06
Netherlands             200128     6492.550  3.419054e+07
Norway                   19247     6529.060  1.350765e+07
Poland                    3653     1422.270  4.341972e+06
Portugal                 16180    13037.540  1.886480e+07
RSA                        352      248.100  7.218680e+05
Saudi Arabia                75       24.110  1.256500e+05
Singapore                 5234    25108.890  2.918376e+06
Spain                    26824    12633.450  3.268929e+07
Sweden                   35637     1806.830  6.790083e+06
Switzerland              30325     6813.690  2.377592e+07
USA                       1034      644.980  3.672086e+06
United Arab Emirates       982      229.890  1.018952e+06
United Kingdom         4263829  2245715.474  5.626433e+09
Unspecified               3300     1204.010  3.348046e+06

In [ ]:
len(set(data['Country']))

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


38

### แปลง data ให้อยู่ในรูป transactions

In [ ]:
data.groupby('InvoiceNo')['StockCode'].apply(list)

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


InvoiceNo
536365     [85123A, 71053, 84406B, 84029G, 84029E, 22752,...
536366                                        [22633, 22632]
536367     [84879, 22745, 22748, 22749, 22310, 84969, 226...
536368                          [22960, 22913, 22912, 22914]
536369                                               [21756]
                                 ...                        
C581484                                              [23843]
C581490                                       [22178, 23144]
C581499                                                  [M]
C581568                                              [21258]
C581569                                       [84978, 20979]
Name: StockCode, Length: 25900, dtype: object

In [ ]:
transac = data.groupby('InvoiceNo')['StockCode'].apply(list)

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


>>Association Rules

In [ ]:
!pip install apyori
from apyori import apriori

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


#Hw1 หา Frequent set item set และ Association rules


*   ของช่วงเวลา Christmas & New year (1 dec - 30 dec)
*   ของประเทศยุโรป
*   ของช่วงเวลาตรุษจีนในทวีป Asia ( 15 Jan - 15 Feb)
*   ของช่วงเวลา Christmas & New year Europe (1 dec - 30 dec)
*   Valentine's day
*   โดยจัดกลุ่มตามประเทศ
groupby ประเทศ - เดือน

In [ ]:
import datetime
import pandas as pd
import os
import pandas as pd
from mlxtend.frequent_patterns import apriori
from mlxtend.frequent_patterns import association_rules
from apyori import apriori

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [ ]:
data['InvoiceDate'] #check InvoiceDate

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


0         12/01/2010 08:26
1         12/01/2010 08:26
2         12/01/2010 08:26
3         12/01/2010 08:26
4         12/01/2010 08:26
                ...       
541904    12/09/2011 12:50
541905    12/09/2011 12:50
541906    12/09/2011 12:50
541907    12/09/2011 12:50
541908    12/09/2011 12:50
Name: InvoiceDate, Length: 541909, dtype: object

In [ ]:
data['InvoiceDate'] = pd.to_datetime(data['InvoiceDate'], format='%m/%d/%Y %H:%M') #วันเดือนปี ชมนาที

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [ ]:
data['Date']=data['InvoiceDate'].dt.date

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [ ]:
data['Date']=pd.to_datetime(data.Date, format='%Y/%m/%d') # แยก ปีเดือนวัน ออกมา

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [ ]:
data['Month']=data['InvoiceDate'].dt.month # แยก เดือน ออกมา

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [ ]:
data

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


InvoiceNo StockCode                          Description  Quantity  \
0         536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1         536365     71053                  WHITE METAL LANTERN         6   
2         536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3         536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4         536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
...          ...       ...                                  ...       ...   
541904    581587     22613          PACK OF 20 SPACEBOY NAPKINS        12   
541905    581587     22899         CHILDREN'S APRON DOLLY GIRL          6   
541906    581587     23254        CHILDRENS CUTLERY DOLLY GIRL          4   
541907    581587     23255      CHILDRENS CUTLERY CIRCUS PARADE         4   
541908    581587     22138        BAKING SET 9 PIECE RETROSPOT          3   

               InvoiceDate  UnitPrice  CustomerID         Country       Date  \
0      2010-12-01 08:26:00       2.55     17850.0  United Kingdom 2010-12-01   
1      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
2      2010-12-01 08:26:00       2.75     17850.0  United Kingdom 2010-12-01   
3      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
4      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
...                    ...        ...         ...             ...        ...   
541904 2011-12-09 12:50:00       0.85     12680.0          France 2011-12-09   
541905 2011-12-09 12:50:00       2.10     12680.0          France 2011-12-09   
541906 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541907 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541908 2011-12-09 12:50:00       4.95     12680.0          France 2011-12-09   

        Month  
0          12  
1          12  
2          12  
3          12  
4          12  
...       ...  
541904     12  
541905     12  
541906     12  
541907     12  
541908     12  

[541909 rows x 10 columns]

## 1. ของช่วงเวลา Christmas & New year (1 dec - 30 dec)

In [ ]:
#1-31 Dec
xmas_new = data[(data['Date'].dt.month == 12) & (data['Date'].dt.day >= 1) & (data['Date'].dt.day <= 31)]
print(xmas_new)

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


       InvoiceNo StockCode                          Description  Quantity  \
0         536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1         536365     71053                  WHITE METAL LANTERN         6   
2         536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3         536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4         536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
...          ...       ...                                  ...       ...   
541904    581587     22613          PACK OF 20 SPACEBOY NAPKINS        12   
541905    581587     22899         CHILDREN'S APRON DOLLY GIRL          6   
541906    581587     23254        CHILDRENS CUTLERY DOLLY GIRL          4   
541907    581587     23255      CHILDRENS CUTLERY CIRCUS PARADE         4   
541908    581587     22138        BAKING SET 9 PIECE RETROSPOT          3   

               InvoiceDate  UnitPrice  CustomerID         Country       Dat

In [ ]:
xmas_new

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


InvoiceNo StockCode                          Description  Quantity  \
0         536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1         536365     71053                  WHITE METAL LANTERN         6   
2         536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3         536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4         536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
...          ...       ...                                  ...       ...   
541904    581587     22613          PACK OF 20 SPACEBOY NAPKINS        12   
541905    581587     22899         CHILDREN'S APRON DOLLY GIRL          6   
541906    581587     23254        CHILDRENS CUTLERY DOLLY GIRL          4   
541907    581587     23255      CHILDRENS CUTLERY CIRCUS PARADE         4   
541908    581587     22138        BAKING SET 9 PIECE RETROSPOT          3   

               InvoiceDate  UnitPrice  CustomerID         Country       Date  \
0      2010-12-01 08:26:00       2.55     17850.0  United Kingdom 2010-12-01   
1      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
2      2010-12-01 08:26:00       2.75     17850.0  United Kingdom 2010-12-01   
3      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
4      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
...                    ...        ...         ...             ...        ...   
541904 2011-12-09 12:50:00       0.85     12680.0          France 2011-12-09   
541905 2011-12-09 12:50:00       2.10     12680.0          France 2011-12-09   
541906 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541907 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541908 2011-12-09 12:50:00       4.95     12680.0          France 2011-12-09   

        Month  
0          12  
1          12  
2          12  
3          12  
4          12  
...       ...  
541904     12  
541905     12  
541906     12  
541907     12  
541908     12  

[68006 rows x 10 columns]

In [ ]:
trans_xmas_new = xmas_new.groupby('InvoiceNo')['Description'].apply(list).reset_index()
print(trans_xmas_new)

     InvoiceNo                                        Description
0       536365  [WHITE HANGING HEART T-LIGHT HOLDER, WHITE MET...
1       536366  [HAND WARMER UNION JACK, HAND WARMER RED POLKA...
2       536367  [ASSORTED COLOUR BIRD ORNAMENT, POPPY'S PLAYHO...
3       536368  [JAM MAKING SET WITH JARS, RED COAT RACK PARIS...
4       536369                         [BATH BUILDING BLOCK WORD]
...        ...                                                ...
3035   C581484                      [PAPER CRAFT , LITTLE BIRDIE]
3036   C581490  [VICTORIAN GLASS HANGING T-LIGHT, ZINC T-LIGHT...
3037   C581499                                           [Manual]
3038   C581568                       [VICTORIAN SEWING BOX LARGE]
3039   C581569  [HANGING HEART JAR T-LIGHT HOLDER, 36 PENCILS ...

[3040 rows x 2 columns]


/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [ ]:
trans_newxmas_apply = trans_xmas_new['Description'].apply(lambda x: [str(item) for item in x])
trans_newxmas_apply

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


0       [WHITE HANGING HEART T-LIGHT HOLDER, WHITE MET...
1       [HAND WARMER UNION JACK, HAND WARMER RED POLKA...
2       [ASSORTED COLOUR BIRD ORNAMENT, POPPY'S PLAYHO...
3       [JAM MAKING SET WITH JARS, RED COAT RACK PARIS...
4                              [BATH BUILDING BLOCK WORD]
                              ...                        
3035                        [PAPER CRAFT , LITTLE BIRDIE]
3036    [VICTORIAN GLASS HANGING T-LIGHT, ZINC T-LIGHT...
3037                                             [Manual]
3038                         [VICTORIAN SEWING BOX LARGE]
3039    [HANGING HEART JAR T-LIGHT HOLDER, 36 PENCILS ...
Name: Description, Length: 3040, dtype: object

In [ ]:
min_sup = 0.036 #เปลี่ยนจนกว่าจะเจอ...

frequentItemsets_newxmas = list(apriori(trans_newxmas_apply, min_support = min_sup))
for i in frequentItemsets_newxmas:
    if i[0].__len__()>1:
        print('yes')
        print(i)

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


yes
RelationRecord(items=frozenset({'ALARM CLOCK BAKELIKE GREEN', 'ALARM CLOCK BAKELIKE RED '}), support=0.03684210526315789, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'ALARM CLOCK BAKELIKE GREEN', 'ALARM CLOCK BAKELIKE RED '}), confidence=0.03684210526315789, lift=1.0), OrderedStatistic(items_base=frozenset({'ALARM CLOCK BAKELIKE GREEN'}), items_add=frozenset({'ALARM CLOCK BAKELIKE RED '}), confidence=0.7088607594936708, lift=14.462662475575565), OrderedStatistic(items_base=frozenset({'ALARM CLOCK BAKELIKE RED '}), items_add=frozenset({'ALARM CLOCK BAKELIKE GREEN'}), confidence=0.7516778523489932, lift=14.462662475575565)])
yes
RelationRecord(items=frozenset({'HOT WATER BOTTLE I AM SO POORLY', 'CHOCOLATE HOT WATER BOTTLE'}), support=0.036513157894736845, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'HOT WATER BOTTLE I AM SO POORLY', 'CHOCOLATE HOT WATER BOTTLE'}), confidence=0.036513157894736845, lift=1.0), O

มี 3 itemset ที่ผ่าน 0.036
1. คนที่ซื้อทั้ง PAPER CHAIN KIT 50'S CHRISTMAS และ PAPER CHAIN KIT VINTAGE CHRISTMAS จาก dataset ทั้งหมด มี 4.77%

   คนที่ซื้อ PAPER CHAIN KIT 50'S CHRISTMAS และ PAPER CHAIN KIT VINTAGE CHRISTMAS มี 4.77%
   
   คนที่ซื้อ PAPER CHAIN KIT 50'S CHRISTMAS แล้วค่อยซื้อ PAPER CHAIN KIT VINTAGE CHRISTMAS มี 51.6%

   คนที่ซื้อ PAPER CHAIN KIT VINTAGE CHRISTMAS แล้วค่อยซื้อ PAPER CHAIN KIT 50'S CHRISTMAS มี 66.51%


2. คนที่ซื้อทั้ง ALARM CLOCK BAKELIKE RED และ ALARM CLOCK BAKELIKE GREEN จาก dataset ทั้งหมด มี 3.68%

   คนที่ซื้อ ALARM CLOCK BAKELIKE RED และ ALARM CLOCK BAKELIKE GREEN มี 3.68%

   คนที่ซื้อ ALARM CLOCK BAKELIKE GREEN แล้วค่อยซื้อ ALARM CLOCK BAKELIKE RED มี 75.17%

   คนที่ซื้อ ALARM CLOCK BAKELIKE RED แล้วค่อยซื้อ ALARM CLOCK BAKELIKE GREEN มี 70.89%



3. คนที่ซื้อ CHOCOLATE HOT WATER BOTTLE และ HOT WATER BOTTLE I AM SO POORLY จาก dataset ทั้งหมด มี 3.65%

   คนที่ซื้อ CHOCOLATE HOT WATER BOTTLE และ HOT WATER BOTTLE I AM SO POORLY มี 3.65%

   คนที่ซื้อ CHOCOLATE HOT WATER BOTTLE แล้วค่อยซื้อ HOT WATER BOTTLE I AM SO POORLY มี 51.63%

   คนที่ซื้อ HOT WATER BOTTLE I AM SO POORLY แล้วค่อยซื้อ CHOCOLATE HOT WATER BOTTLE มี 68.94%



##2. ของประเทศยุโรป

##จัดกลุ่มประเทศตามทวีป

In [ ]:
set(data['Country']) #check country

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


{'Australia',
 'Austria',
 'Bahrain',
 'Belgium',
 'Brazil',
 'Canada',
 'Channel Islands',
 'Cyprus',
 'Czech Republic',
 'Denmark',
 'EIRE',
 'European Community',
 'Finland',
 'France',
 'Germany',
 'Greece',
 'Hong Kong',
 'Iceland',
 'Israel',
 'Italy',
 'Japan',
 'Lebanon',
 'Lithuania',
 'Malta',
 'Netherlands',
 'Norway',
 'Poland',
 'Portugal',
 'RSA',
 'Saudi Arabia',
 'Singapore',
 'Spain',
 'Sweden',
 'Switzerland',
 'USA',
 'United Arab Emirates',
 'United Kingdom',
 'Unspecified'}

In [ ]:
Eur = data[(data['Country']!='Austria') | (data['Country']!='Belgium')| (data['Country']!='Channel Islands') | (data['Country']!='Cyprus') | (data['Country']!='Czech Republic') | (data['Country']!='Denmark') |
           (data['Country']!='EIRE') | (data['Country']!='European Community')| (data['Country']!='Finland')| (data['Country']!='France') | (data['Country']!='Germany')| (data['Country']!='Greece')|
           (data['Country']!='Iceland') | (data['Country']!='Italy') | (data['Country']!='Lithuania') | (data['Country']!='Malta') | (data['Country']!='Netherlands') | (data['Country']!='Norway') |
           (data['Country']!='Poland') | (data['Country']!='Portugal') | (data['Country']!='Spain') | (data['Country']!='Sweden') | (data['Country']!='Switzerland') | (data['Country']!='United Kingdom')]
Eur #แบ่งทวีปยุโรป

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


InvoiceNo StockCode                          Description  Quantity  \
0         536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1         536365     71053                  WHITE METAL LANTERN         6   
2         536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3         536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4         536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
...          ...       ...                                  ...       ...   
541904    581587     22613          PACK OF 20 SPACEBOY NAPKINS        12   
541905    581587     22899         CHILDREN'S APRON DOLLY GIRL          6   
541906    581587     23254        CHILDRENS CUTLERY DOLLY GIRL          4   
541907    581587     23255      CHILDRENS CUTLERY CIRCUS PARADE         4   
541908    581587     22138        BAKING SET 9 PIECE RETROSPOT          3   

               InvoiceDate  UnitPrice  CustomerID         Country       Date  \
0      2010-12-01 08:26:00       2.55     17850.0  United Kingdom 2010-12-01   
1      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
2      2010-12-01 08:26:00       2.75     17850.0  United Kingdom 2010-12-01   
3      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
4      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
...                    ...        ...         ...             ...        ...   
541904 2011-12-09 12:50:00       0.85     12680.0          France 2011-12-09   
541905 2011-12-09 12:50:00       2.10     12680.0          France 2011-12-09   
541906 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541907 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541908 2011-12-09 12:50:00       4.95     12680.0          France 2011-12-09   

        Month  
0          12  
1          12  
2          12  
3          12  
4          12  
...       ...  
541904     12  
541905     12  
541906     12  
541907     12  
541908     12  

[541909 rows x 10 columns]

In [ ]:
trans_Eur = Eur.groupby('InvoiceNo')['Description'].apply(list).reset_index()
trans_Europe = trans_Eur['Description'].apply(lambda x: [str(item) for item in x])

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [ ]:
trans_Europe

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


0        [WHITE HANGING HEART T-LIGHT HOLDER, WHITE MET...
1        [HAND WARMER UNION JACK, HAND WARMER RED POLKA...
2        [ASSORTED COLOUR BIRD ORNAMENT, POPPY'S PLAYHO...
3        [JAM MAKING SET WITH JARS, RED COAT RACK PARIS...
4                               [BATH BUILDING BLOCK WORD]
                               ...                        
25895                        [PAPER CRAFT , LITTLE BIRDIE]
25896    [VICTORIAN GLASS HANGING T-LIGHT, ZINC T-LIGHT...
25897                                             [Manual]
25898                         [VICTORIAN SEWING BOX LARGE]
25899    [HANGING HEART JAR T-LIGHT HOLDER, 36 PENCILS ...
Name: Description, Length: 25900, dtype: object

In [ ]:
min_sup = 0.03

frequentItemsets_Europe = list(apriori(trans_Europe, min_support = min_sup))
for i in frequentItemsets_Europe:
    if i[0].__len__()>1:
        print('yes')
        print(i)

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


yes
RelationRecord(items=frozenset({'GREEN REGENCY TEACUP AND SAUCER', 'ROSES REGENCY TEACUP AND SAUCER '}), support=0.03027027027027027, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'GREEN REGENCY TEACUP AND SAUCER', 'ROSES REGENCY TEACUP AND SAUCER '}), confidence=0.03027027027027027, lift=1.0), OrderedStatistic(items_base=frozenset({'GREEN REGENCY TEACUP AND SAUCER'}), items_add=frozenset({'ROSES REGENCY TEACUP AND SAUCER '}), confidence=0.7417218543046358, lift=17.1523178807947), OrderedStatistic(items_base=frozenset({'ROSES REGENCY TEACUP AND SAUCER '}), items_add=frozenset({'GREEN REGENCY TEACUP AND SAUCER'}), confidence=0.7, lift=17.1523178807947)])
yes
RelationRecord(items=frozenset({'JUMBO BAG RED RETROSPOT', 'JUMBO BAG PINK POLKADOT'}), support=0.03216216216216216, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'JUMBO BAG RED RETROSPOT', 'JUMBO BAG PINK POLKADOT'}), confidence=0.03216216216216216, lift=1.

มี 2 itemset ที่ผ่าน 0.03

1. คนที่ซื้อทั้ง jumbo bag red retrospot และ jumbo bag pink polkadot จาก dataset ทั้งหมด มี 3.22%

   คนที่ซื้อ JUMBO BAG RED RETROSPOT และ JUMBO BAG PINK POLKADOT มี 3.22%

   คนที่ซื้อ JUMBO BAG PINK POLKADOT แล้วค่อยซื้อ JUMBO BAG RED RETROSPOT มี 67.67%

   คนที่ซื้อ JUMBO BAG RED RETROSPOT แล้วค่อยซื้อ JUMBO BAG PINK POLKADOT มี 39.02%

2. คนที่ซื้อทั้ง green regency teacup and saucer และ roses regency teacup and saucer จาก dataset ทั้งหมด มี 3.03%

   คนที่ซื้อ GREEN REGENCY TEACUP AND SAUCER และ ROSES REGENCY TEACUP AND SAUCER มี 3.03%

   คนที่ซื้อ GREEN REGENCY TEACUP AND SAUCER แล้วค่อยซื้อ ROSES REGENCY TEACUP AND SAUCER มี 74.17%

   คนที่ซื้อ ROSES REGENCY TEACUP AND SAUCER แล้วค่อยซื้อ GREEN REGENCY TEACUP AND SAUCER มี 70.00%

## 3. ของช่วงเวลาตรุษจีนในทวีป Asia ( 15 Jan - 15 Feb)

In [ ]:
Asia = data[(data['Country']=='Bahrain') | (data['Country']=='Hong Kong')| (data['Country']=='Singapore') | (data['Country']=='Japan') | (data['Country']=='Israel') |
            (data['Country']=='Lebanon') | (data['Country']=='Saudi Arabia') | (data['Country']=='United Arab Emirates')]
Asia #แบ่งทวีปเอเชีย

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


InvoiceNo StockCode                     Description  Quantity  \
9783      537218     85016   SET OF 6 VINTAGE NOTELETS KIT         6   
9784      537218     21506      FANCY FONT BIRTHDAY CARD,         12   
9785      537218     22716              CARD CIRCUS PARADE        12   
9786      537218     22983             CARD BILLBOARD FONT        12   
9787      537218     22030          SWALLOWS GREETING CARD        12   
...          ...       ...                             ...       ...   
530276   C580832     21731   RED TOADSTOOL LED NIGHT LIGHT        -2   
530277   C580832     23480  MINI LIGHTS WOODLAND MUSHROOMS       -15   
530278   C580832     22466  FAIRY TALE COTTAGE NIGHT LIGHT       -17   
530279   C580832     23084              RABBIT NIGHT LIGHT        -7   
530280   C580832     22621      TRADITIONAL KNITTING NANCY        -1   

               InvoiceDate  UnitPrice  CustomerID Country       Date  Month  
9783   2010-12-05 15:46:00       2.55     12763.0   Japan 2010-12-05     12  
9784   2010-12-05 15:46:00       0.42     12763.0   Japan 2010-12-05     12  
9785   2010-12-05 15:46:00       0.42     12763.0   Japan 2010-12-05     12  
9786   2010-12-05 15:46:00       0.42     12763.0   Japan 2010-12-05     12  
9787   2010-12-05 15:46:00       0.42     12763.0   Japan 2010-12-05     12  
...                    ...        ...         ...     ...        ...    ...  
530276 2011-12-06 11:40:00       1.45     12753.0   Japan 2011-12-06     12  
530277 2011-12-06 11:40:00       3.29     12753.0   Japan 2011-12-06     12  
530278 2011-12-06 11:40:00       1.65     12753.0   Japan 2011-12-06     12  
530279 2011-12-06 11:40:00       1.79     12753.0   Japan 2011-12-06     12  
530280 2011-12-06 11:40:00       1.45     12753.0   Japan 2011-12-06     12  

[1314 rows x 10 columns]

In [ ]:
#Chinese Newyear - Lunar
Lunar = Asia[(Asia['Date'].dt.month == 1) & (Asia['Date'].dt.day >= 15) & (data['Date'].dt.day <= 31)  |
                   ((Asia['Date'].dt.month == 2) & (Asia['Date'].dt.day <= 15))]
print(Lunar)

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


      InvoiceNo StockCode                          Description  Quantity  \
69623    541975     22620          4 TRADITIONAL SPINNING TOPS        60   
69624    541975    85183B  CHARLIE & LOLA WASTEPAPER BIN FLORA         6   
69625    541975    85183A   CHARLIE & LOLA WASTEPAPER BIN BLUE         6   
69626    541975     16236                 KITTY PENCIL ERASERS        28   
69627    541975     21058               PARTY INVITES WOODLAND        12   
...         ...       ...                                  ...       ...   
89595    543911    47590B          PINK HAPPY BIRTHDAY BUNTING        12   
89596    543911     22117     METAL SIGN HER DINNER IS SERVED         12   
89597    543911     82567            AIRLINE LOUNGE,METAL SIGN        12   
89598    543911     21910                  WAY OUT METAL SIGN         12   
89599    543911     21911                   GARDEN METAL SIGN         12   

              InvoiceDate  UnitPrice  CustomerID               Country  \
69623 2011-01

<ipython-input-345-c497d0e7be5b>:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  Lunar = Asia[(Asia['Date'].dt.month == 1) & (Asia['Date'].dt.day >= 15) & (data['Date'].dt.day <= 31)  |


In [ ]:
trans_lunar = Lunar.groupby('InvoiceNo')['Description'].apply(list).reset_index()
trans_lunar_apply = trans_lunar['Description'].apply(lambda x: [str(item) for item in x])

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [ ]:
trans_lunar_apply

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


0    [4 TRADITIONAL SPINNING TOPS, CHARLIE & LOLA W...
1    [GIN & TONIC DIET GREETING CARD , GREEN REGENC...
2    [LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT MET...
3    [MONEY BOX FIRST ADE DESIGN, MONEY BOX KINGS C...
4    [SET OF 3 NOTEBOOKS IN PARCEL, LUNCH BAG DOLLY...
5    [RETROSPOT HEART HOT WATER BOTTLE, DOORMAT UNI...
Name: Description, dtype: object

In [ ]:
min_sup = 0.33

frequentItemset_China = list(apriori(trans_lunar_apply, min_support = min_sup))
for i in frequentItemset_China:
    if i[0].__len__()>1:
        print('yes')
        print(i)

yes
RelationRecord(items=frozenset({'BASKET OF TOADSTOOLS', 'LUNCH BAG RED RETROSPOT'}), support=0.3333333333333333, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'BASKET OF TOADSTOOLS', 'LUNCH BAG RED RETROSPOT'}), confidence=0.3333333333333333, lift=1.0), OrderedStatistic(items_base=frozenset({'BASKET OF TOADSTOOLS'}), items_add=frozenset({'LUNCH BAG RED RETROSPOT'}), confidence=1.0, lift=3.0), OrderedStatistic(items_base=frozenset({'LUNCH BAG RED RETROSPOT'}), items_add=frozenset({'BASKET OF TOADSTOOLS'}), confidence=1.0, lift=3.0)])
yes
RelationRecord(items=frozenset({'SET 3 RETROSPOT TEA,COFFEE,SUGAR', 'BASKET OF TOADSTOOLS'}), support=0.3333333333333333, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'SET 3 RETROSPOT TEA,COFFEE,SUGAR', 'BASKET OF TOADSTOOLS'}), confidence=0.3333333333333333, lift=1.0), OrderedStatistic(items_base=frozenset({'BASKET OF TOADSTOOLS'}), items_add=frozenset({'SET 3 RETROSPOT TEA,CO

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


มี 44 itemset ที่ผ่าน 0.03

1. คนที่ซื้อทั้ง BASKET OF TOADSTOOLS และ LUNCH BAG RED RETROSPOT จาก dataset ทั้งหมด มี 33.33%

   คนที่ซื้อ BASKET OF TOADSTOOLS และ LUNCH BAG RED RETROSPOT มี 33.33%

   คนที่ซื้อ BASKET OF TOADSTOOLS แล้วค่อยซื้อ LUNCH BAG RED RETROSPOT มี 100.00%

   คนที่ซื้อ LUNCH BAG RED RETROSPOT แล้วค่อยซื้อ BASKET OF TOADSTOOLS มี 100.00%

--------------------------------------------------------------------------------

2. คนที่ซื้อทั้ง SET 3 RETROSPOT TEA,COFFEE,SUGAR และ BASKET OF TOADSTOOLS จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ RETROSPOT TEA,COFFEE,SUGAR และ BASKET OF TOADSTOOLS มี 33.33%

  คนที่ซื้อ BASKET OF TOADSTOOLS แล้วค่อยซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR มี 100.00%

  คนที่ซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR แล้วค่อยซื้อ BASKET OF TOADSTOOLS มี 100.00%

--------------------------------------------------------------------------------

3. คนที่ซื้อทั้ง JAM MAKING SET WITH JARS และ BATH BUILDING BLOCK WORD จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ JAM MAKING SET WITH JARS และ BATH BUILDING BLOCK WORD มี 33.33%

  คนที่ซื้อ BATH BUILDING BLOCK WORD แล้วค่อยซื้อ JAM MAKING SET WITH JARS มี 100.00%

  คนที่ซื้อ SET 3 JAM MAKING SET WITH JARS แล้วค่อยซื้อ BATH BUILDING BLOCK WORD มี 100.00%

--------------------------------------------------------------------------------

4. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN และ BATH BUILDING BLOCK WORD จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN และ BATH BUILDING BLOCK WORD มี 33.33%

  คนที่ซื้อ BATH BUILDING BLOCK WORD แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ SET 3 LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ BATH BUILDING BLOCK WORD มี 66.67%

--------------------------------------------------------------------------------

5. คนที่ซื้อทั้ง SET 3 PAPER VINTAGE CHICK PAPER EGG และ BATH BUILDING BLOCK WORD จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG และ BATH BUILDING BLOCK WORD มี 33.33%

  คนที่ซื้อ BATH BUILDING BLOCK WORD แล้วค่อยซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ BATH BUILDING BLOCK WORD มี 66.67%

--------------------------------------------------------------------------------

6. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN และ BEWARE OF THE CAT METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN และ BEWARE OF THE CAT METAL SIGN มี 33.33%

  คนที่ซื้อ BEWARE OF THE CAT METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ BEWARE OF THE CAT METAL SIGN มี 66.67%

--------------------------------------------------------------------------------

7. คนที่ซื้อทั้ง LAUNDRY 15C METAL SIGN และ BEWARE OF THE CAT METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN และ BEWARE OF THE CAT METAL SIGN มี 33.33%

  คนที่ซื้อ BEWARE OF THE CAT METAL SIGN แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ BEWARE OF THE CAT METAL SIGN มี 100.00%

--------------------------------------------------------------------------------

8. คนที่ซื้อทั้ง WASHROOM METAL SIGN และ BEWARE OF THE CAT METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ WASHROOM METAL SIGN และ BEWARE OF THE CAT METAL SIGN มี 33.33%

  คนที่ซื้อ BEWARE OF THE CAT METAL SIGN แล้วค่อยซื้อ WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN แล้วค่อยซื้อ BEWARE OF THE CAT METAL SIGN มี 100.00%

--------------------------------------------------------------------------------

9. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN และ DOORMAT UNION FLAG จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN และ DOORMAT UNION FLAG มี 33.33%

  คนที่ซื้อ DOORMAT UNION FLAG แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ DOORMAT UNION FLAG มี 66.67%

--------------------------------------------------------------------------------

10. คนที่ซื้อทั้ง DOORMAT UNION FLAG และ METAL SIGN TAKE IT OR LEAVE IT จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ DOORMAT UNION FLAG และ METAL SIGN TAKE IT OR LEAVE IT มี 33.33%

  คนที่ซื้อ DOORMAT UNION FLAG แล้วค่อยซื้อ METAL SIGN TAKE IT OR LEAVE IT มี 100.00%

  คนที่ซื้อ METAL SIGN TAKE IT OR LEAVE IT แล้วค่อยซื้อ DOORMAT UNION FLAG มี 100.00%

--------------------------------------------------------------------------------

11. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN และ JAM MAKING SET WITH JARS จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN และ JAM MAKING SET WITH JARS มี 33.33%

  คนที่ซื้อ JAM MAKING SET WITH JARS แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ JAM MAKING SET WITH JARS มี 66.67%

--------------------------------------------------------------------------------

12. คนที่ซื้อทั้ง JAM MAKING SET WITH JARS และ SET 3 PAPER VINTAGE CHICK PAPER EGG จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ JAM MAKING SET WITH JARS และ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 33.33%

  คนที่ซื้อ JAM MAKING SET WITH JARS แล้วค่อยซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ JAM MAKING SET WITH JARS มี 66.67%

--------------------------------------------------------------------------------

13. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN และ LAUNDRY 15C METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN และ LAUNDRY 15C METAL SIGN มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN มี 66.67%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

--------------------------------------------------------------------------------

14. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN และ METAL SIGN TAKE IT OR LEAVE IT จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN และ METAL SIGN TAKE IT OR LEAVE IT มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ METAL SIGN TAKE IT OR LEAVE IT มี 66.67%

  คนที่ซื้อ METAL SIGN TAKE IT OR LEAVE IT แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

--------------------------------------------------------------------------------

15. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN และ SET 3 PAPER VINTAGE CHICK PAPER EGG จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN และ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 66.67%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 66.67%

--------------------------------------------------------------------------------

16. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN และ WASHROOM METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN และ WASHROOM METAL SIGN มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ WASHROOM METAL SIGN มี 66.67%

  คนที่ซื้อ WASHROOM METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

--------------------------------------------------------------------------------

17. คนที่ซื้อทั้ง WASHROOM METAL SIGN และ LAUNDRY 15C METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ WASHROOM METAL SIGN และ LAUNDRY 15C METAL SIGN มี 33.33%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN มี 100.00%

--------------------------------------------------------------------------------

18. คนที่ซื้อทั้ง SET 3 RETROSPOT TEA,COFFEE,SUGAR และ LUNCH BAG RED RETROSPOT จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR และ LUNCH BAG RED RETROSPOT มี 33.33%

  คนที่ซื้อ LUNCH BAG RED RETROSPOT แล้วค่อยซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR มี 100.00%

  คนที่ซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR แล้วค่อยซื้อ LUNCH BAG RED RETROSPOT มี 100.00%

--------------------------------------------------------------------------------

19. คนที่ซื้อทั้ง MILK PAN PINK POLKADOT และ MILK PAN RED RETROSPOT จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT และ MILK PAN RED RETROSPOT มี 33.33%

  คนที่ซื้อ MILK PAN RED RETROSPOT แล้วค่อยซื้อ MILK PAN PINK POLKADOT มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT แล้วค่อยซื้อ MILK PAN RED RETROSPOT มี 100.00%

--------------------------------------------------------------------------------

20. คนที่ซื้อทั้ง MILK PAN PINK POLKADOT และ PLASTERS IN TIN SPACEBOY จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT และ PLASTERS IN TIN SPACEBOY มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN PINK POLKADOT มี 100.00%

--------------------------------------------------------------------------------

21. คนที่ซื้อทั้ง MILK PAN PINK POLKADOT และ RED RETROSPOT CHILDRENS UMBRELLA จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT และ RED RETROSPOT CHILDRENS UMBRELLA ทั้งหมด มี 33.33% มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT แล้วค่อยซื้อ RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN PINK POLKADOT มี 100.00%

--------------------------------------------------------------------------------

22. คนที่ซื้อทั้ง MILK PAN RED RETROSPOT และ PLASTERS IN TIN SPACEBOY จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN RED RETROSPOT และ PLASTERS IN TIN SPACEBOY มี 33.33%

  คนที่ซื้อ MILK PAN RED RETROSPOT แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN RED RETROSPOT มี 100.00%

--------------------------------------------------------------------------------

23. คนที่ซื้อทั้ง MILK PAN RED RETROSPOT และ RED RETROSPOT CHILDRENS UMBRELLA จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN RED RETROSPOT และ RED RETROSPOT CHILDRENS UMBRELLA มี 33.33%

  คนที่ซื้อ MILK PAN RED RETROSPOT แล้วค่อยซื้อ RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN RED RETROSPOT มี 100.00%

--------------------------------------------------------------------------------

24. คนที่ซื้อทั้ง PARTY BUNTING และ SET 3 PAPER VINTAGE CHICK PAPER EGG จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ PARTY BUNTING และ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 33.33%

  คนที่ซื้อ PARTY BUNTING แล้วค่อยซื้อ PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ PARTY BUNTING มี 66.67%

--------------------------------------------------------------------------------

25. คนที่ซื้อทั้ง PLASTERS IN TIN SPACEBOY และ RED RETROSPOT CHILDRENS UMBRELLA EGG จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY และ RED RETROSPOT CHILDRENS UMBRELLA EGG มี 33.33%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ RED RETROSPOT CHILDRENS UMBRELLA EGG มี 100.00%

  คนที่ซื้อ RED RETROSPOT CHILDRENS UMBRELLA EGG แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY มี 100.00%

--------------------------------------------------------------------------------

26. คนที่ซื้อทั้ง SET OF 72 PINK HEART PAPER DOILIES และ SET 3 PAPER VINTAGE CHICK PAPER EGG จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG และ SET OF 72 PINK HEART PAPER DOILIES มี 33.33%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ SET OF 72 PINK HEART PAPER DOILIES มี 66.67%

  คนที่ซื้อ SET OF 72 PINK HEART PAPER DOILIES แล้วค่อยซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

--------------------------------------------------------------------------------

27. คนที่ซื้อทั้ง SET OF 72 PINK HEART PAPER DOILIES และ SET 3 PAPER VINTAGE CHICK PAPER EGG จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG และ SET OF 72 PINK HEART PAPER DOILIES มี 33.33%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ SET OF 72 PINK HEART PAPER DOILIES มี 66.67%

  คนที่ซื้อ SET OF 72 PINK HEART PAPER DOILIES แล้วค่อยซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

--------------------------------------------------------------------------------

28. คนที่ซื้อทั้ง SET 3 RETROSPOT TEA,COFFEE,SUGAR, BASKET OF TOADSTOOLS, LUNCH BAG RED RETROSPOT จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR, BASKET OF TOADSTOOLS, LUNCH BAG RED RETROSPOT มี 33.33%

  คนที่ซื้อ BASKET OF TOADSTOOLS แล้วค่อยซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR, LUNCH BAG RED RETROSPOT มี 100.00%

  คนที่ซื้อ LUNCH BAG RED RETROSPOT แล้วค่อยซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR, BASKET OF TOADSTOOLS มี 100.00%

  คนที่ซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR แล้วค่อยซื้อ BASKET OF TOADSTOOLS, LUNCH BAG RED RETROSPOT มี 100.00%

  คนที่ซื้อ BASKET OF TOADSTOOLS, LUNCH BAG RED RETROSPOT แล้วค่อยซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR มี 100.00%

  คนที่ซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR, BASKET OF TOADSTOOLS แล้วค่อยซื้อ LUNCH BAG RED RETROSPOT มี 100.00%

  คนที่ซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR, LUNCH BAG RED RETROSPOT แล้วค่อยซื้อ BASKET OF TOADSTOOLS มี 100.00%

--------------------------------------------------------------------------------

29. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD มี 33.33%

  คนที่ซื้อ BATH BUILDING BLOCK WORD แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS มี 100.00%

  คนที่ซื้อ JAM MAKING SET WITH JARS แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, BATH BUILDING BLOCK WORD มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD มี 66.67%

  คนที่ซื้อ JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ JAM MAKING SET WITH JARS มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS แล้วค่อยซื้อ BATH BUILDING BLOCK WORD มี 100.00%

--------------------------------------------------------------------------------

30. คนที่ซื้อทั้ง JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG, BATH BUILDING BLOCK WORD จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 33.33%

  คนที่ซื้อ BATH BUILDING BLOCK WORD แล้วค่อยซื้อ JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ JAM MAKING SET WITH JARS แล้วค่อยซื้อ BATH BUILDING BLOCK WORD, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD มี 66.67%

  คนที่ซื้อ JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ JAM MAKING SET WITH JARS มี 100.00%

  คนที่ซื้อ JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ BATH BUILDING BLOCK WORD มี 100.00%

--------------------------------------------------------------------------------

31. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN, SET 3 PAPER VINTAGE CHICK PAPER EGG, BATH BUILDING BLOCK WORD จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, BATH BUILDING BLOCK WORD, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 33.33%

  คนที่ซื้อ BATH BUILDING BLOCK WORD แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ BATH BUILDING BLOCK WORD, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 66.67%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, BATH BUILDING BLOCK WORD มี 66.67%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ BATH BUILDING BLOCK WORD มี 100.00%

--------------------------------------------------------------------------------

32. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 33.33%

  คนที่ซื้อ BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 66.67%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ BEWARE OF THE CAT METAL SIGN มี 100.00%

--------------------------------------------------------------------------------

33. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 66.67%

  คนที่ซื้อ WASHROOM METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN แล้วค่อยซื้อ BEWARE OF THE CAT METAL SIGN  มี 100.00%

--------------------------------------------------------------------------------

34. คนที่ซื้อทั้ง WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 33.33%

  คนที่ซื้อ BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 100.00%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ BEWARE OF THE CAT METAL SIGN  มี 100.00%

--------------------------------------------------------------------------------

35. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN, DOORMAT UNION FLAG, METAL SIGN TAKE IT OR LEAVE IT จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, DOORMAT UNION FLAG, METAL SIGN TAKE IT OR LEAVE IT  มี 33.33%

  คนที่ซื้อ DOORMAT UNION FLAG แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, METAL SIGN TAKE IT OR LEAVE IT  มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ DOORMAT UNION FLAG, METAL SIGN TAKE IT OR LEAVE IT  มี 66.67%

  คนที่ซื้อ METAL SIGN TAKE IT OR LEAVE IT  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, DOORMAT UNION FLAG มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, DOORMAT UNION FLAG แล้วค่อยซื้อ METAL SIGN TAKE IT OR LEAVE IT  มี 100.00%

  คนที่ซื้อ DOORMAT UNION FLAG, METAL SIGN TAKE IT OR LEAVE IT  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, METAL SIGN TAKE IT OR LEAVE IT  แล้วค่อยซื้อ DOORMAT UNION FLAG มี 100.00%

--------------------------------------------------------------------------------

36. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 33.33%

  คนที่ซื้อ JAM MAKING SET WITH JARS แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 66.67%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS มี 66.67%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS แล้วค่อยซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ JAM MAKING SET WITH JARS มี 100.00%

--------------------------------------------------------------------------------

37. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, WASHROOM METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, WASHROOM METAL SIGN มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN มี 66.67%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

--------------------------------------------------------------------------------

38. คนที่ซื้อทั้ง MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT แล้วค่อยซื้อ MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT แล้วค่อยซื้อ MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN RED RETROSPOT มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN PINK POLKADOT มี 100.00%

--------------------------------------------------------------------------------

39. คนที่ซื้อทั้ง MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT แล้วค่อยซื้อ MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT แล้วค่อยซื้อ MILK PAN PINK POLKADOT, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT แล้วค่อยซื้อ RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN RED RETROSPOT มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN PINK POLKADOT มี 100.00%

--------------------------------------------------------------------------------

40. คนที่ซื้อทั้ง MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN PINK POLKADOT, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN PINK POLKADOT มี 100.00%

--------------------------------------------------------------------------------

41. คนที่ซื้อทั้ง MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA มี 33.33%

  คนที่ซื้อ MILK PAN RED RETROSPOT แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN RED RETROSPOT มี 100.00%

--------------------------------------------------------------------------------

42. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG, BATH BUILDING BLOCK WORD จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG, BATH BUILDING BLOCK WORD มี 33.33%

  คนที่ซื้อ BATH BUILDING BLOCK WORD แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ JAM MAKING SET WITH JARS แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, BATH BUILDING BLOCK WORD, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 66.67%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD มี 66.67%

  คนที่ซื้อ JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS แล้วค่อยซื้อ BATH BUILDING BLOCK WORD, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, BATH BUILDING BLOCK WORD มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, SET 3 PAPER VINTAGE CHICK PAPER EGG, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ JAM MAKING SET WITH JARS มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ BATH BUILDING BLOCK WORD มี 100.00%

--------------------------------------------------------------------------------

43. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 33.33%

  คนที่ซื้อ BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 66.67%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, WASHROOM METAL SIGN แล้วค่อยซื้อ BEWARE OF THE CAT METAL SIGN  มี 100.00%

--------------------------------------------------------------------------------

44. คนที่ซื้อทั้ง MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT แล้วค่อยซื้อ MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT แล้วค่อยซื้อ MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN PINK POLKADOT, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN RED RETROSPOT มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN PINK POLKADOT มี 100.00%

## 4. ของช่วงเวลา Christmas & New year Europe (1 dec - 30 dec)


In [ ]:
Eur

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


InvoiceNo StockCode                          Description  Quantity  \
0         536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1         536365     71053                  WHITE METAL LANTERN         6   
2         536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3         536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4         536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
...          ...       ...                                  ...       ...   
541904    581587     22613          PACK OF 20 SPACEBOY NAPKINS        12   
541905    581587     22899         CHILDREN'S APRON DOLLY GIRL          6   
541906    581587     23254        CHILDRENS CUTLERY DOLLY GIRL          4   
541907    581587     23255      CHILDRENS CUTLERY CIRCUS PARADE         4   
541908    581587     22138        BAKING SET 9 PIECE RETROSPOT          3   

               InvoiceDate  UnitPrice  CustomerID         Country       Date  \
0      2010-12-01 08:26:00       2.55     17850.0  United Kingdom 2010-12-01   
1      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
2      2010-12-01 08:26:00       2.75     17850.0  United Kingdom 2010-12-01   
3      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
4      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
...                    ...        ...         ...             ...        ...   
541904 2011-12-09 12:50:00       0.85     12680.0          France 2011-12-09   
541905 2011-12-09 12:50:00       2.10     12680.0          France 2011-12-09   
541906 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541907 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541908 2011-12-09 12:50:00       4.95     12680.0          France 2011-12-09   

        Month  
0          12  
1          12  
2          12  
3          12  
4          12  
...       ...  
541904     12  
541905     12  
541906     12  
541907     12  
541908     12  

[541909 rows x 10 columns]

In [ ]:
#Europe Christmas & New year
Eur_newxmas = Eur[(Eur['Date'].dt.month == 12) & (Eur['Date'].dt.day >= 1) & (Eur['Date'].dt.day <= 31)]
Eur_newxmas

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


InvoiceNo StockCode                          Description  Quantity  \
0         536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1         536365     71053                  WHITE METAL LANTERN         6   
2         536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3         536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4         536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
...          ...       ...                                  ...       ...   
541904    581587     22613          PACK OF 20 SPACEBOY NAPKINS        12   
541905    581587     22899         CHILDREN'S APRON DOLLY GIRL          6   
541906    581587     23254        CHILDRENS CUTLERY DOLLY GIRL          4   
541907    581587     23255      CHILDRENS CUTLERY CIRCUS PARADE         4   
541908    581587     22138        BAKING SET 9 PIECE RETROSPOT          3   

               InvoiceDate  UnitPrice  CustomerID         Country       Date  \
0      2010-12-01 08:26:00       2.55     17850.0  United Kingdom 2010-12-01   
1      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
2      2010-12-01 08:26:00       2.75     17850.0  United Kingdom 2010-12-01   
3      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
4      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
...                    ...        ...         ...             ...        ...   
541904 2011-12-09 12:50:00       0.85     12680.0          France 2011-12-09   
541905 2011-12-09 12:50:00       2.10     12680.0          France 2011-12-09   
541906 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541907 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541908 2011-12-09 12:50:00       4.95     12680.0          France 2011-12-09   

        Month  
0          12  
1          12  
2          12  
3          12  
4          12  
...       ...  
541904     12  
541905     12  
541906     12  
541907     12  
541908     12  

[68006 rows x 10 columns]

In [ ]:
trans2_newxmas = Eur_newxmas.groupby('InvoiceNo')['Description'].apply(list).reset_index()
trans2_newxmas_apply = trans2_newxmas['Description'].apply(lambda x: [str(item) for item in x])

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [ ]:
trans2_newxmas_apply

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


0       [WHITE HANGING HEART T-LIGHT HOLDER, WHITE MET...
1       [HAND WARMER UNION JACK, HAND WARMER RED POLKA...
2       [ASSORTED COLOUR BIRD ORNAMENT, POPPY'S PLAYHO...
3       [JAM MAKING SET WITH JARS, RED COAT RACK PARIS...
4                              [BATH BUILDING BLOCK WORD]
                              ...                        
3035                        [PAPER CRAFT , LITTLE BIRDIE]
3036    [VICTORIAN GLASS HANGING T-LIGHT, ZINC T-LIGHT...
3037                                             [Manual]
3038                         [VICTORIAN SEWING BOX LARGE]
3039    [HANGING HEART JAR T-LIGHT HOLDER, 36 PENCILS ...
Name: Description, Length: 3040, dtype: object

In [ ]:
min_sup = 0.036

frequentItemset_newxmas = list(apriori(trans2_newxmas_apply, min_support = min_sup))
for i in frequentItemset_newxmas:
    if i[0].__len__()>1:
        print('yes')
        print(i)

yes
RelationRecord(items=frozenset({'ALARM CLOCK BAKELIKE GREEN', 'ALARM CLOCK BAKELIKE RED '}), support=0.03684210526315789, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'ALARM CLOCK BAKELIKE GREEN', 'ALARM CLOCK BAKELIKE RED '}), confidence=0.03684210526315789, lift=1.0), OrderedStatistic(items_base=frozenset({'ALARM CLOCK BAKELIKE GREEN'}), items_add=frozenset({'ALARM CLOCK BAKELIKE RED '}), confidence=0.7088607594936708, lift=14.462662475575565), OrderedStatistic(items_base=frozenset({'ALARM CLOCK BAKELIKE RED '}), items_add=frozenset({'ALARM CLOCK BAKELIKE GREEN'}), confidence=0.7516778523489932, lift=14.462662475575565)])
yes
RelationRecord(items=frozenset({'HOT WATER BOTTLE I AM SO POORLY', 'CHOCOLATE HOT WATER BOTTLE'}), support=0.036513157894736845, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'HOT WATER BOTTLE I AM SO POORLY', 'CHOCOLATE HOT WATER BOTTLE'}), confidence=0.036513157894736845, lift=1.0), O

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


มี 3 itemset ที่ผ่าน 0.035
1. คนที่ซื้อทั้ง PAPER CHAIN KIT 50'S CHRISTMAS และ PAPER CHAIN KIT VINTAGE CHRISTMAS จาก dataset ทั้งหมด มี 4.77%

   คนที่ซื้อ PAPER CHAIN KIT 50'S CHRISTMAS และ PAPER CHAIN KIT VINTAGE CHRISTMAS  มี 4.77%
   
   คนที่ซื้อ PAPER CHAIN KIT 50'S CHRISTMAS แล้วค่อยซื้อ PAPER CHAIN KIT VINTAGE CHRISTMAS มี 51.6%

   คนที่ซื้อ PAPER CHAIN KIT VINTAGE CHRISTMAS แล้วค่อยซื้อ PAPER CHAIN KIT 50'S CHRISTMAS มี 66.51%


2. คนที่ซื้อทั้ง ALARM CLOCK BAKELIKE GREEN และ ALARM CLOCK BAKELIKE RED จาก dataset ทั้งหมด มี 3.68%

   คนที่ซื้อ ALARM CLOCK BAKELIKE RED และ ALARM CLOCK BAKELIKE GREEN มี 3.68%

   คนที่ซื้อ ALARM CLOCK BAKELIKE GREEN แล้วค่อยซื้อ ALARM CLOCK BAKELIKE RED มี 70.89%

   คนที่ซื้อ ALARM CLOCK BAKELIKE RED แล้วค่อยซื้อ ALARM CLOCK BAKELIKE GREEN มี 75.17%



3. คนที่ซื้อ CHOCOLATE HOT WATER BOTTLE และ HOT WATER BOTTLE I AM SO POORLY จาก dataset ทั้งหมด มี 3.65%

   คนที่ซื้อ CHOCOLATE HOT WATER BOTTLE และ HOT WATER BOTTLE I AM SO POORLY มี 3.65%

   คนที่ซื้อ CHOCOLATE HOT WATER BOTTLE แล้วค่อยซื้อ HOT WATER BOTTLE I AM SO POORLY มี 51.63%

   คนที่ซื้อ HOT WATER BOTTLE I AM SO POORLY แล้วค่อยซื้อ CHOCOLATE HOT WATER BOTTLE มี 68.94%


## 5. Valentine's day


In [ ]:
# 14 Feb - rose
rose = data[(data['Date'].dt.month == 2) & (data['Date'].dt.day ==14)]
print(rose)

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


      InvoiceNo StockCode                         Description  Quantity  \
88847    543821     22926      IVORY GIANT GARDEN THERMOMETER         2   
88848    543821     22928     YELLOW GIANT GARDEN THERMOMETER         2   
88849    543821     22925       BLUE GIANT GARDEN THERMOMETER         2   
88850    543821     85175               CACTI T-LIGHT CANDLES        16   
88851    543821     22456     NATURAL SLATE CHALKBOARD LARGE          3   
...         ...       ...                                 ...       ...   
89950    543979     21889              WOODEN BOX OF DOMINOES         1   
89951    543979     85176          SEWING SUSAN 21 NEEDLE SET         1   
89952    543979     37446  MINI CAKE STAND WITH HANGING CAKES         1   
89953    543979     21657     MILK BOTTLE WITH GLASS STOPPER          1   
89954   C543980     22728           ALARM CLOCK BAKELIKE PINK        -1   

              InvoiceDate  UnitPrice  CustomerID         Country       Date  \
88847 2011-02-14 09:

In [ ]:
rose

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


InvoiceNo StockCode                         Description  Quantity  \
88847    543821     22926      IVORY GIANT GARDEN THERMOMETER         2   
88848    543821     22928     YELLOW GIANT GARDEN THERMOMETER         2   
88849    543821     22925       BLUE GIANT GARDEN THERMOMETER         2   
88850    543821     85175               CACTI T-LIGHT CANDLES        16   
88851    543821     22456     NATURAL SLATE CHALKBOARD LARGE          3   
...         ...       ...                                 ...       ...   
89950    543979     21889              WOODEN BOX OF DOMINOES         1   
89951    543979     85176          SEWING SUSAN 21 NEEDLE SET         1   
89952    543979     37446  MINI CAKE STAND WITH HANGING CAKES         1   
89953    543979     21657     MILK BOTTLE WITH GLASS STOPPER          1   
89954   C543980     22728           ALARM CLOCK BAKELIKE PINK        -1   

              InvoiceDate  UnitPrice  CustomerID         Country       Date  \
88847 2011-02-14 09:24:00       5.95     17667.0  United Kingdom 2011-02-14   
88848 2011-02-14 09:24:00       5.95     17667.0  United Kingdom 2011-02-14   
88849 2011-02-14 09:24:00       5.95     17667.0  United Kingdom 2011-02-14   
88850 2011-02-14 09:24:00       0.42     17667.0  United Kingdom 2011-02-14   
88851 2011-02-14 09:24:00       4.95     17667.0  United Kingdom 2011-02-14   
...                   ...        ...         ...             ...        ...   
89950 2011-02-14 15:43:00       1.25     14606.0  United Kingdom 2011-02-14   
89951 2011-02-14 15:43:00       0.85     14606.0  United Kingdom 2011-02-14   
89952 2011-02-14 15:43:00       1.45     14606.0  United Kingdom 2011-02-14   
89953 2011-02-14 15:43:00       6.95     14606.0  United Kingdom 2011-02-14   
89954 2011-02-14 16:39:00       3.75     14189.0  United Kingdom 2011-02-14   

       Month  
88847      2  
88848      2  
88849      2  
88850      2  
88851      2  
...      ...  
89950      2  
89951      2  
89952      2  
89953      2  
89954      2  

[1108 rows x 10 columns]

In [ ]:
trans_rose = rose.groupby('InvoiceNo')['Description'].apply(list).reset_index()
trans_rose_apply = trans_rose['Description'].apply(lambda x: [str(item) for item in x])

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [ ]:
trans_rose_apply

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


0     [IVORY GIANT GARDEN THERMOMETER, YELLOW GIANT ...
1     [RETROSPOT TEA SET CERAMIC 11 PC , 12 PENCIL S...
2     [BAKING SET 9 PIECE RETROSPOT , GLASS  BEURRE ...
3     [POPPY'S PLAYHOUSE BEDROOM , POPPY'S PLAYHOUSE...
4     [IVORY GIANT GARDEN THERMOMETER, SET 3 WICKER ...
5     [PURPLE FRANGIPANI HAIRCLIP, FINE SILVER NECKL...
6                       [incorrectly made-thrown away.]
7     [SET OF 6 HERB TINS SKETCHBOOK, SET OF 6 SPICE...
8     [RED RETROSPOT CUP, PINK  POLKADOT CUP, BLUE P...
9     [TOMATO CHARLIE+LOLA COASTER SET, JUMBO BAG TO...
10    [ASSORTED COLOUR BIRD ORNAMENT, 6 RIBBONS RUST...
11    [BROCANTE COAT RACK, SET 3 PAPER VINTAGE CHICK...
12    [SET OF 36 TEATIME PAPER DOILIES, 60 TEATIME F...
13    [FRYING PAN UNION FLAG, RED TOADSTOOL LED NIGH...
14                 [GROW A FLYTRAP OR SUNFLOWER IN TIN]
15    [CHARLOTTE BAG SUKI DESIGN, JUMBO STORAGE BAG ...
16    [CARRIAGE, CLASSIC METAL BIRDCAGE PLANT HOLDER...
17    [CHARLOTTE BAG SUKI DESIGN, WOODLAND CHARL

In [ ]:
min_sup = 0.09

frequentItemsets_rose = list(apriori(trans_rose_apply, min_support = min_sup))
for i in frequentItemsets_rose:
    if i[0].__len__()>1:
        print('yes')
        print(i)

yes
RelationRecord(items=frozenset({'SET OF 3 CAKE TINS PANTRY DESIGN ', 'HOMEMADE JAM SCENTED CANDLES'}), support=0.09090909090909091, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'SET OF 3 CAKE TINS PANTRY DESIGN ', 'HOMEMADE JAM SCENTED CANDLES'}), confidence=0.09090909090909091, lift=1.0), OrderedStatistic(items_base=frozenset({'HOMEMADE JAM SCENTED CANDLES'}), items_add=frozenset({'SET OF 3 CAKE TINS PANTRY DESIGN '}), confidence=1.0, lift=6.875), OrderedStatistic(items_base=frozenset({'SET OF 3 CAKE TINS PANTRY DESIGN '}), items_add=frozenset({'HOMEMADE JAM SCENTED CANDLES'}), confidence=0.625, lift=6.875)])
yes
RelationRecord(items=frozenset({'SET OF 3 CAKE TINS PANTRY DESIGN ', 'SET OF 6 SPICE TINS PANTRY DESIGN'}), support=0.09090909090909091, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'SET OF 3 CAKE TINS PANTRY DESIGN ', 'SET OF 6 SPICE TINS PANTRY DESIGN'}), confidence=0.09090909090909091, lift=1.0),

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


มี 2 itemset ที่ผ่าน 0.09
1. คนที่ซื้อทั้ง SET OF 3 CAKE TINS PANTRY DESIGN และ HOMEMADE JAM SCENTED CANDLES จาก dataset ทั้งหมด มี 9.09%

   คนที่ซื้อ SET OF 3 CAKE TINS PANTRY DESIGN และ HOMEMADE JAM SCENTED CANDLES  มี 9.09%
   
   คนที่ซื้อ HOMEMADE JAM SCENTED CANDLES แล้วค่อยซื้อ SET OF 3 CAKE TINS PANTRY DESIGN มี 100%

   คนที่ซื้อ SET OF 3 CAKE TINS PANTRY DESIGN  แล้วค่อยซื้อ HOMEMADE JAM SCENTED CANDLES มี 62.5%


2. คนที่ซื้อทั้ง SET OF 3 CAKE TINS PANTRY DESIGN และ SET OF 6 SPICE TINS PANTRY DESIGN ทั้งหมด มี 9.09%

   คนที่ซื้อ SET OF 3 CAKE TINS PANTRY DESIGN และ SET OF 6 SPICE TINS PANTRY DESIGN มี 9.09%

   คนที่ซื้อ SET OF 3 CAKE TINS PANTRY DESIGN  แล้วค่อยซื้อ SET OF 6 SPICE TINS PANTRY DESIGN มี 62.5%

   คนที่ซื้อ SET OF 6 SPICE TINS PANTRY DESIGN แล้วค่อยซื้อ SET OF 3 CAKE TINS PANTRY DESIGN มี 83.33%

## 6. โดยจัดกลุ่มตามประเทศ groupby ประเทศ - เดือน

In [ ]:
data

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


InvoiceNo StockCode                          Description  Quantity  \
0         536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1         536365     71053                  WHITE METAL LANTERN         6   
2         536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3         536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4         536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
...          ...       ...                                  ...       ...   
541904    581587     22613          PACK OF 20 SPACEBOY NAPKINS        12   
541905    581587     22899         CHILDREN'S APRON DOLLY GIRL          6   
541906    581587     23254        CHILDRENS CUTLERY DOLLY GIRL          4   
541907    581587     23255      CHILDRENS CUTLERY CIRCUS PARADE         4   
541908    581587     22138        BAKING SET 9 PIECE RETROSPOT          3   

               InvoiceDate  UnitPrice  CustomerID         Country       Date  \
0      2010-12-01 08:26:00       2.55     17850.0  United Kingdom 2010-12-01   
1      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
2      2010-12-01 08:26:00       2.75     17850.0  United Kingdom 2010-12-01   
3      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
4      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
...                    ...        ...         ...             ...        ...   
541904 2011-12-09 12:50:00       0.85     12680.0          France 2011-12-09   
541905 2011-12-09 12:50:00       2.10     12680.0          France 2011-12-09   
541906 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541907 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541908 2011-12-09 12:50:00       4.95     12680.0          France 2011-12-09   

        Month  
0          12  
1          12  
2          12  
3          12  
4          12  
...       ...  
541904     12  
541905     12  
541906     12  
541907     12  
541908     12  

[541909 rows x 10 columns]

In [ ]:
trans_CM = data.groupby(['Country','Month'])['Description'].apply(list).reset_index()
trans_Country_M = trans_CM['Description'].apply(lambda x: [str(item) for item in x])
trans_Country_M

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


0      [RIBBON REEL LACE DESIGN , RIBBON REEL HEARTS ...
1      [HANGING HEART JAR T-LIGHT HOLDER, JAM MAKING ...
2      [PANTRY ROLLING PIN, PANTRY SCRUBBING BRUSH, S...
3      [POSTAGE, JIGSAW TOADSTOOLS 3 PIECE, SET OF 4 ...
4      [ROSES REGENCY TEACUP AND SAUCER , EASTER TIN ...
                             ...                        
291    [PARTY CHARMS 50 PIECES, PACK OF 6 LARGE FRUIT...
292    [ASSORTED MINI MADRAS NOTEBOOK, PACK OF 72 RET...
293    [STORAGE TIN VINTAGE DOILY , SET OF 72 RETROSP...
294    [CALENDAR PAPER CUT DESIGN, DINOSAURS  WRITING...
295    [ROUND SNACK BOXES SET OF4 WOODLAND , LUNCH BA...
Name: Description, Length: 296, dtype: object

In [ ]:
min_sup = 0.31

frequentItemset_CM = list(apriori(trans_Country_M, min_support = min_sup))
for i in frequentItemset_CM:
    if i[0].__len__()>1:
        print('yes')
        print(i)

yes
RelationRecord(items=frozenset({'POSTAGE', 'REGENCY CAKESTAND 3 TIER'}), support=0.3141891891891892, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'POSTAGE', 'REGENCY CAKESTAND 3 TIER'}), confidence=0.3141891891891892, lift=1.0), OrderedStatistic(items_base=frozenset({'POSTAGE'}), items_add=frozenset({'REGENCY CAKESTAND 3 TIER'}), confidence=0.5470588235294118, lift=1.190657439446367), OrderedStatistic(items_base=frozenset({'REGENCY CAKESTAND 3 TIER'}), items_add=frozenset({'POSTAGE'}), confidence=0.6838235294117647, lift=1.1906574394463667)])
yes
RelationRecord(items=frozenset({'POSTAGE', 'ROUND SNACK BOXES SET OF4 WOODLAND '}), support=0.32094594594594594, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'POSTAGE', 'ROUND SNACK BOXES SET OF4 WOODLAND '}), confidence=0.32094594594594594, lift=1.0), OrderedStatistic(items_base=frozenset({'POSTAGE'}), items_add=frozenset({'ROUND SNACK BOXES SET OF4 WOODLAND '}), co

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


มี 2 itemset ที่ผ่าน 0.31
1. คนที่ซื้อทั้ง POSTAGE และ ROUND SNACK BOXES SET OF4 WOODLAND จาก dataset ทั้งหมด มี 32.09%

   คนที่ซื้อ POSTAGE และ ROUND SNACK BOXES SET OF4 WOODLAN มี 32.09%
  
   คนที่ซื้อ POSTAGE แล้วค่อยซื้อ ROUND SNACK BOXES SET OF4 WOODLAND มี 55.88%

   คนที่ซื้อ ROUND SNACK BOXES SET OF4 WOODLAND แล้วค่อยซื้อ POSTAGE มี 84.07%

2. คนที่ซื้อทั้ง POSTAGE และ REGENCY CAKESTAND 3 TIER จาก dataset ทั้งหมด มี 31.41%

   คนที่ซื้อ POSTAGE และ REGENCY CAKESTAND 3 TIER มี 31.41%
  
   คนที่ซื้อ POSTAGE แล้วค่อยซื้อ REGENCY CAKESTAND 3 TIER มี 54.70%

   คนที่ซื้อ REGENCY CAKESTAND 3 TIER แล้วค่อยซื้อ POSTAGE มี 68.38%
